cdap-data-fabric/src/main/java/io/cdap/cdap/data2/datafabric/dataset/RemoteDatasetFramework.java

```diff
@@ -158,37 +163,39 @@ public void updateInstance(DatasetId datasetInstanceId, DatasetProperties props)
   @Override
   public Collection<DatasetSpecificationSummary> getInstances(NamespaceId namespaceId)
     throws DatasetManagementException {
-    return clientCache.getUnchecked(namespaceId).getAllInstances();
+    return Retries.callWithRetries(() -> clientCache.getUnchecked(namespaceId).getAllInstances(), retryStrategy);
   }
 
   @Override
   public Collection<DatasetSpecificationSummary> getInstances(NamespaceId namespaceId, Map<String, String> properties)
     throws DatasetManagementException {
-    return clientCache.getUnchecked(namespaceId).getInstances(properties);
+    return Retries.callWithRetries(() -> clientCache.getUnchecked(namespaceId).getInstances(properties), retryStrategy);
```

Comment:
Does the `clientCache.getUnchecked` populate `RetryableException` correctly? If the getUnchecked wraps exception, then the `callWithRetries` may not work as intended.

Reply:
done